Std/Data/List/Lemmas.lean

```diff
@@ -1063,6 +1063,22 @@ theorem Pairwise.imp {α R S} (H : ∀ {a b}, R a b → S a b) :
   | _, .nil => .nil
   | _, .cons h₁ h₂ => .cons (H ∘ h₁ ·) (h₂.imp H)
 
+theorem pairwise_append_comm_of_symm {α R} (s : ∀ {a b}, R a b → R b a) {l₁ l₂ : List α} :
+    Pairwise R (l₁ ++ l₂) ↔ Pairwise R (l₂ ++ l₁) := by
+  have : ∀ l₁ l₂ : List α, (∀ x : α, x ∈ l₁ → ∀ y : α, y ∈ l₂ → R x y) →
+    ∀ x : α, x ∈ l₂ → ∀ y : α, y ∈ l₁ → R x y := fun l₁ l₂ a x xm y ym => s (a y ym x xm)
+  have and_left_comm : ∀ a b c, (a ∧ b ∧ c) = (b ∧ a ∧ c) := fun a b c => by
```

Comment:
@digama0 note that because we don't have the commutativity typeclasses, in particular we don't have [this lemma](https://github.com/leanprover-community/mathlib4/blob/master/Mathlib/Init/Logic.lean#L351) so I had to prove it inline.